python/cudf/cudf/core/column/column.py

```diff
@@ -1273,44 +1273,79 @@ def scatter_to_table(
             }
         )
 
-    def _copy_type_metadata(self: T, other: ColumnBase) -> ColumnBase:
+    @property
+    def type_metadata(self):
+        """
+        Return metadata relevant for constructing a copy of this column.
+
+        The metadata will always contain the dtype of the column, in addition
+        to:
+
+        * the categories and ordering, if ``self`` is a ``CategoricalColumn``
+        * the field names, if ``self`` is a ``StructColumn``
+        * the precision, if ``self`` is a ``DecimalColumn``
+        """
+        metadata = {"type": self.__class__}
+
+        if isinstance(self, cudf.core.column.CategoricalColumn):
+            metadata.update(
+                {"categories": self.categories, "ordered": self.ordered}
+            )
+        if isinstance(self, cudf.core.column.StructColumn):
+            metadata["field_keys"] = self.dtype.fields.keys()
+        if isinstance(self, cudf.core.column.DecimalColumn):
+            metadata["precision"] = self.dtype.precision
```

Comment:
For Structs / Decimals the info is already attached to the dtype here. For Categoricals I believe the categories are attached to the dtype as well, where we should always have the info to reconstruct a column given its dtype, no?

Reply:
That's correct, but wouldn't we need access to `self` to do that?

The main function of these changes would be that we could make a copy of a column that is not in scope - for example, when unpacking a `PackedColumns` object from #8153, we need to apply the correct dtype metadata to each resulting column, but we probably don't want to have to access the original packed DataFrame to do this.

With these changes, we could store each columns' `type_metadata` in the `PackedColumns` object, and use that for reconstruction instead.

Reply:
I guess my point is why do we need a new function associated with the Column class to do this when all of the information already exists on the dtype? I.E. I believe we already have `__serialize__` and `__deserialize__` implemented for all of our dtypes, so we should already be able to do `serialize(column.dtype)` to shove into the `PackedColumns` object.

Reply:
That makes sense - in that case, I think a good roadmap would be to add the serialize/deserialize methods to the `StructDtype` so that all the types affected here are serializable, and then implement the relevant `as_*_column()` methods so that we can cast the columns with the dtype alone.